crmsh/sh.py

```diff
@@ -0,0 +1,321 @@
+"""Run shell commands.
+
+This module provides various methods to run shell commands, on both local and remote hosts, as current or another user.
+There many variant of the methods to allow fine-gain control of parameter passing and error handling.
+
+3 different implementations are provided:
+
+1. LocalShell allows to run command on local host as various users. It is the most feature-rich one, allowing
+   interactive I/O from/to the terminal.
+2. SshShell allows to run command on both local and remote hosts. When running on a remote host, it creates a direct
+   connection from a specified local_user to the destination host and user. User input from terminal is not allowed, as
+   the command is passed through the stdin.
+3. AutoShell runs command on cluster nodes. It leverages su, sudo and ssh to obtain a appreciated session on a
+   destination node. It is only available after ssh bootstrap as it depends on the knowledge about cluster node and user
+   configurations.
```

Comment:
Since this kind of shell is `only available after ssh bootstrap` and it's for the cluster nodes, how about renaming it as `ClusterShell`?